core/app/[locale]/(default)/(auth)/change-password/page.tsx

```diff
@@ -22,17 +25,18 @@ export async function generateMetadata(): Promise<Metadata> {
   };
 }
 
-export default async function ChangePassword({ searchParams }: Props) {
-  const { c: customerEntityId, t: token } = await searchParams;
+export default async function ChangePassword({ params, searchParams }: Props) {
+  const { locale } = await params;
+  const { c: customerEntityId, t: token } = await loadSearchParams(searchParams);
   const t = await getTranslations('ChangePassword');
-  const locale = await getLocale();
 
   if (!customerEntityId || !token) {
     return redirect({ href: '/login', locale });
   }
```

Comment:
This could be moved to middleware but then we'd be splitting business logic in two places, which is not great.